### ISSUE TITLE
[Simplex_tree] Rules of 5 is violated

### ISSUE THREAD

**Issue by VincentRouvreau** [GUDHI/gudhi-devel#22]
Missing move and copy assignement.

```cpp
    Simplex_tree& operator=(const Simplex_tree& other) // copy assignment
    {
         ...;
    }
    Simplex_tree& operator=(Simplex_tree&& other) noexcept // move assignment
    {
        ...;
    }
```

**Comment by VincentRouvreau** [comment]
To be reviewed.

Branch: svn+ssh://scm.gforge.inria.fr/svnroot/gudhi/branches/simplex_tree_fix_vincent

Modified/New files:
* src/Simplex_tree/include/gudhi/Simplex_tree.h
* src/Simplex_tree/test/CMakeLists.txt
* src/Simplex_tree/test/simplex_tree_ctor_and_move_unit_test.cpp

**Comment by VincentRouvreau** [comment]
New Revision: 3977

Log:
Merge branch simplex_tree_fix_vincent to fix issue #22 ([Simplex_tree] Rules of 5 is violated) and #23 ([Simplex_tree] shallow move constructor)
